__tests__/app.test.js

```diff
@@ -442,3 +442,116 @@ describe("GET /api/users", () => {
+                .expect(200)
+                .then(({ body }) => {
+                    const { articles } = body;
+
+                    expect(articles).toBeInstanceOf(Array);
+                    expect(articles).toHaveLength(13);
+                    expect(articles).toBeSortedBy("author", { descending: true });
+
+                    articles.forEach((article) => {
+                        expect(article).toMatchObject({
+                            author: expect.any(String),
+                            title: expect.any(String),
+                            article_id: expect.any(Number),
+                            topic: expect.any(String),
+                            created_at: expect.any(String),
+                            votes: expect.any(Number),
+                            article_img_url: expect.any(String),
+                            comment_count: expect.any(String),
+                        });
+                    });
+                });
+        });
+
+        test("400: should respond with 'Bad request' if provided with an incorrect sort_by value", () => {
+            return request(app)
+                .get("/api/articles?sort_by=nonsense")
+                .expect(400)
+                .then(({ body }) => {
+                    expect(body.msg).toBe("Bad request");
+                });
+        });
+    });
+
+    describe("GET /api/articles?order=asc", () => {
+        test("200: should respond with an array of all articles sorted by date in ascending order, the default order is descending", () => {
+            return request(app)
+                .get("/api/articles?order=asc")
+                .expect(200)
+                .then(({ body }) => {
+                    const { articles } = body;
+
+                    expect(articles).toBeInstanceOf(Array);
+                    expect(articles).toHaveLength(13);
+                    expect(articles).toBeSortedBy("created_at");
+
+                    articles.forEach((article) => {
+                        expect(article).toMatchObject({
+                            author: expect.any(String),
+                            title: expect.any(String),
+                            article_id: expect.any(Number),
+                            topic: expect.any(String),
+                            created_at: expect.any(String),
+                            votes: expect.any(Number),
+                            article_img_url: expect.any(String),
+                            comment_count: expect.any(String),
+                        });
+                    });
+                });
+        });
+
+        test("400: should respond with 'Bad request' if provided with an incorrect order value", () => {
+            return request(app)
+                .get("/api/articles?order=anna")
+                .expect(400)
+                .then(({ body }) => {
+                    expect(body.msg).toBe("Bad request");
+                });
+        });
+    });
```

Comment:
The topic 'paper' is a valid topic, but it has no articles attached to it! What would happen in that scenario?